frontend/src/pages/auth/components/fields/name/NameField.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {useEffect, useState} from 'react';
+import {NameUserCredentialsProps} from "./props";
+import {ValidationErrorMessages} from "../../../../../common/constansts/validationErrorMessages";
+import {TextField} from "@mui/material";
+import {TEXT_FIELD_VALIDATION} from "../../../../../common/constansts/textFieldValidation";
+
+const NameField: React.FC<NameUserCredentialsProps> = ({
+                                                           defaultName,
+                                                           updateNameFields,
+                                                       }: NameUserCredentialsProps) => {
+
+    const [name, setName] = useState<string>(defaultName);
+    const [isNameValid, setIsNameValid] = useState<boolean>(false);
+
+    const handleNameChange = (name: string) => {
+        setName(name);
+        const isValid = validateName(name);
+        setIsNameValid(isValid);
+        updateNameFields(name, isValid);
+    };
+
+    const validateName = (name: string) => name.trim().length >= TEXT_FIELD_VALIDATION;
+
+    useEffect(() => {
+        setName(defaultName);
+        validateName(defaultName);
+    }, []);
+
+    return (
+        <TextField
+            error={!isNameValid}
+            helperText={!isNameValid ? ValidationErrorMessages.NAME_NOT_VALID : ""}
```

Comment:
```suggestion
            helperText={isNameValid ? "" : ValidationErrorMessages.NAME_NOT_VALID}
```

If we can reduce amount of inversions, conditions -> it's easier to read